src/main/kotlin/com/openlattice/linking/FeedbackType.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2019. OpenLattice, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@openlattice.com
+ *
+ *
+ */
+
+package com.openlattice.linking
+
+enum class FeedbackType {
```

Comment:
Is this necessary if both link and unlink are distinguishing whether feedback is positive/negative?

Reply:
It is for getting all feedback of 1 entity: either you request only positive, negative or all feedbacks.
We could change it to always return all feedback and then filtering what we want, but I think its faster to use the mapstore index to do that

Reply:
If you have 1 entity I feel like there are three case all unlink, all link, and some link+some unlink. Those would seem to correspond the three FeedbackType classes you enumerated. Am I missing something here?

Reply:
You are not returning a LinkingFeedback object, but a collection of EntityLinkingFeedbacks, which are consisting of an EntityKeyPair and a boolean